lisa/sut_orchestrator/azure/features.py

```diff
@@ -293,6 +292,12 @@ def check(self, capability: Any) -> search_space.ResultReason:
             ),
             "data_disk_size",
         )
+        result.merge(
```

Comment:
You can consider to use a SetSpace to support the usage here. Req can be one of True/False/None, and cap can be a set of {True, False}, True, False.

Reply:
SetSpace doesn't feel right her as the value could only be one of {True, False, None}. I will go with your earlier suggestion of keeping the logic in DiskOptionSettings .